### ISSUE TITLE
Maps images and graphs for email

### ISSUE THREAD

**Issue by mushon** [mushon/IWBA#22]
Ideally it would be great to have the email come with small maps and graphs projecting the revenue. Something like this:
![iwba_email](https://cloud.githubusercontent.com/assets/27123/19796936/7d6e3398-9cf0-11e6-83d6-ecb4405e6e80.png)

With the limits of the email system we're using and the limits of the technology, would that be feasible? And if not, what would?

**Comment by wonyoungso** [comment]
Graph and map cannot be dynamically generated--you know, since it's email. To make this feasible, we need to manually add images of a map of spots and projection graphs and so on. Embedding webfonts can be possible in some mail clients; 

https://litmus.com/blog/typography-tips-for-email-designers

**Comment by wonyoungso** [comment]
And projection data also should be added in spots table in administration page.

**Comment by wonyoungso** [comment]
mail template can be modified in `/app/views/deposit_mailer/deposit_mail.html.erb`

**Comment by mushon** [comment]
Yes, I know the maps and graphs should be rendered on the server and served in the email. Mapbox has an API for rendering static image maps from GL. Either way, we can always prepare the hotspots in advance (basically create screenshots manually). 
My main question is whether you think this is something we should do, or is it too much work at this point. Otherwise I'm open to other solutions. 

Mushon Zer-Aviv
Mushon.com | Shual.com | @Mushon

**Comment by wonyoungso** [comment]
I think showing water projection data of invested spots will strengthen our works little bit because through pouring experience we can know what's happening around the world about the water. But I don't know..crafting email is basically annoying task so..

**Comment by mushon** [comment]
We can pretender the hotspots and basically also pretender the percentile projections, then just link to pre-rendered images, no?

**Comment by wonyoungso** [comment]
Yes, if you make some of the graphs and maps of hotspots and send them to me, then I can upload them to somewhere in cloud and point them.

**Comment by mushon** [comment]
Ok. The hotspots will be easier than the graphs, but maybe I can think of something. Maybe we can use bars instead and render them in simple HTML. 

Mushon Zer-Aviv
Mushon.com | Shual.com | @mushon

**Comment by wonyoungso** [comment]
Rendering graph in email natively may be challenging although it's simple, because in some email clients they misinterpret `<div>` tag and all of css should be inline. We need to take simple approach..

**Comment by mushon** [comment]
@wonyoungso can we simply pre-render all of the maps and graphs (in percentage) as PNGs on the server and then include them in the email?

**Comment by mushon** [comment]
Image URL structure:
`https://api.mapbox.com/styles/v1/mushon/cittuq85x000s2irqfdmd1kdr/static/127.523067,36.346416,5.46,0.00,0.00/600x400?access_token=pk.eyJ1IjoibXVzaG9uIiwiYSI6IjY1bHhhTkEifQ.DhW2zcurHHBtmnc2FsMBqg`
![](https://api.mapbox.com/styles/v1/mushon/cittuq85x000s2irqfdmd1kdr/static/127.523067,36.346416,5.46,0.00,0.00/600x400?access_token=pk.eyJ1IjoibXVzaG9uIiwiYSI6IjY1bHhhTkEifQ.DhW2zcurHHBtmnc2FsMBqg)

**Comment by mushon** [comment]
Pointers for server-side rendering of d3 code:
https://gist.github.com/vicapow/758fce6aa4c5195d24be
http://stackoverflow.com/questions/11632748/svg-to-png-server-side-using-node-js?answertab=votes#tab-top
https://gist.github.com/mattbaker/1509145

**Comment by wonyoungso** [comment]
images are done, but I seriously think it is feasible in serverside rendering in graph..

**Comment by mushon** [comment]
Not sure I understood your previous response. So you think it IS feasible? Or not?

Mushon Zer-Aviv
Mushon.com | Shual.com | @Mushon

**Comment by wonyoungso** [comment]
not feasible. (oh I wrote in completely opposite way..) it's heavy since it should be done in serverside and I need to install nodejs thing in heroku server and it's gonna be messed up. maybe we'll do it in just simple <div> bar graph..

**Comment by wonyoungso** [comment]
except that, almost done.

**Comment by mushon** [comment]
Fair enough. How about this bar graph:

![image](https://cloud.githubusercontent.com/assets/27123/20029079/500704f0-a34b-11e6-8fce-33d396f15606.png)

**Comment by mushon** [comment]
after @wonyoungso implements #33 @mushon will generate static images for graphs, push them to assets/images and reference them into the admin

**Comment by wonyoungso** [comment]
check https://github.com/mushon/IWBA/issues/33#issuecomment-259041604
